### DISCUSSION TITLE
Why support GFM tables?

### DISCUSSION THREAD

**Discussion by emmatown** [markdoc/markdoc#387] [Q&A] [answered]
The [docs for the `table` tag](https://markdoc.dev/docs/tags#table) reference "CommonMark tables" but tables don't exist in CommonMark, they're [an extension from GitHub-flavoured Markdown](https://github.github.com/gfm/#tables-extension-) so since they're not in CommonMark itself and Markdoc has its own table syntax, why support GFM tables? Especially when Markdoc already does things like not supporting setext headings to avoid having two syntaxes?

**Comment by rpaul-stripe** [comment, marked as answer]
To be honest, I think the only reason we initially supported GFM tables is because we had a lot of them in our legacy documentation at Stripe and didn't want to convert them all during our migration to Markdoc. The GFM table syntax is objectively bad and imposes a lot of arbitrary limitations on the content that can be nested inside. I am open to eventually phasing them out, especially now that the formatter is maturing and making it easier to automatically convert them into Markdoc tables. 

We could provide more opinionated guidance in the docs to make it clear that GFM table support is intended for legacy content and that new content written today should use the Markdoc table syntax. And then in a future major release of Markdoc where we comfortable making breaking changes we could make them optional and default them to off.

**Reply by emmatown** [reply]
Cool! that makes a lot of sense, good to know I'm not missing something
